drivers/staging/iio/adc/ad7192.c

```
@@ -622,63 +643,38 @@ static const struct iio_chan_spec ad7193_channels[] = {
 	IIO_CHAN_SOFT_TIMESTAMP(14),
 };
 
-static int ad7192_parse_dt(struct device_node *np,
-               struct ad7192_platform_data *pdata)
+static void ad7192_parse_dt(struct device_node *np,
+               struct ad7192_state *st)
 {
-	of_property_read_u16(np, "adi,reference-voltage-mv", &pdata->vref_mv);
-	of_property_read_u8(np, "adi,clock-source-select", &pdata->clock_source_sel);
-	of_property_read_u32(np, "adi,external-clock-Hz", &pdata->ext_clk_hz);
-	pdata->refin2_en = of_property_read_bool(np, "adi,refin2-pins-enable");
-	pdata->rej60_en = of_property_read_bool(np, "adi,rejection-60-Hz-enable");
-	pdata->sinc3_en = of_property_read_bool(np, "adi,sinc3-filter-enable");
-	pdata->chop_en = of_property_read_bool(np, "adi,chop-enable");
-	pdata->buf_en = of_property_read_bool(np, "adi,buffer-enable");
-	pdata->unipolar_en = of_property_read_bool(np, "adi,unipolar-enable");
-	pdata->burnout_curr_en = of_property_read_bool(np, "adi,burnout-currents-enable");
-
-	return 0;
+	st->refin2_en = of_property_read_bool(np, "adi,refin2-pins-enable");
+	st->rej60_en = of_property_read_bool(np, "adi,rejection-60-Hz-enable");
+	st->sinc3_en = of_property_read_bool(np, "adi,sinc3-filter-enable");
+	st->chop_en = of_property_read_bool(np, "adi,chop-enable");
+	st->buf_en = of_property_read_bool(np, "adi,buffer-enable");
+	st->unipolar_en = of_property_read_bool(np, "adi,unipolar-enable");
+	st->burnout_curr_en = of_property_read_bool(np, "adi,burnout-currents-enable");
```

Since all this properties are required, you also need to return an error in case they are missing.

---

These are all optional. No return needed.

---

Ok. However, some of these properties do not belong in the dt binding. 
* refin2-pins-enable: you can add a second regulator in the device tree. If it exists, then you can enable refin2.
* rejection-60-Hz-enable and sinc3-filter-enable. You should use the IIO_CHAN_INFO_LOW_PASS_FILTER_3DB_FREQUENCY attribute to set the bw.
* buffer-enable: we can use the iio_triggered_buffer_setup().
* adi,unipolar-enable: it's probably fine, but soon a generic binding called `bipolar `will be added under Documentation/devicetree/bindings/iio/adc/adc.txt. You can follow this discussion: https://lore.kernel.org/patchwork/patch/1011859/.
* about adi,chop-enable and adi,burnout-currents-enable I am not sure, but you can try to check other bindings and see if there is anything similar.

---

Some of the above remarks haven't been addressed.

---

are you referring to the unipolar option ?

---

Yes, unipolar should be replaced with bipolar. Also, I think that you do no need `refin2-pins-enable`, `rejection-60-Hz-enable`  and` sinc3-filter-enable` anymore, since you found a better approach. Have you checked the other properties? To me they look like configuration options that should be left to user space.